.github/workflows/build.yml

```diff
@@ -0,0 +1,95 @@
+name: "Build"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  id-token: write
+  contents: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          cache: yarn
+          node-version-file: .nvmrc
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: corretto
+          cache: sbt
+          java-version: 11
+
+      - run: make reinstall
+
+      - name: assets test
+        run: ./tools/task-runner/runner validate test
+
+      - name: assets compile
+        run: ./tools/task-runner/runner compile
+        env:
+          NODE_ENV: production
+
+      - uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: eu-west-1
+          role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
+
+      - name: Test, compile
+        run: |
+          java \
+          -Xmx6144M \
+          -XX:ReservedCodeCacheSize=128m \
+          -Dsbt.log.noformat=true \
+          -XX:+UseParallelGC \
+          -DAPP_SECRET="fake_secret" \
+          -Duser.timezone=Australia/Sydney \
+          -jar ./bin/sbt-launch.jar clean compile assets scalafmtCheckAll test Universal/packageBin
+        env:
+          STAGE: DEVINFRA
```

Comment:
Good q, my assumption is that it's the stage for running automated tests in CI/CD. The `config` module specifically does things differently for `DEVINFRA`, like using a properties file instead of looking values up from parameter store [^1]. Although the name `DEVINFRA` isn't clear, I think it's what we want for running tests in GHA. I'm confused why it wasn't being used in TC.

[^1]: https://github.com/guardian/frontend/blob/c8db82bdf6b188350e0b5a843960422bb6b4d332/common/app/common/configuration.scala#L88

Reply:
Another option here is to set the `DEVINFRA` stage by default for running tests from sbt. I don't think there's a case where we'd _want_ to try and load live config from Parameter Store (which depends on valid AWS credentials in the environment) when running unit tests?

Reply:
I've gone with the second option here, to set `DEVINFRA` by default